python-checks/src/main/java/org/sonar/python/checks/SklearnEstimatorHyperparametersCheck.java

```diff
@@ -0,0 +1,178 @@
+    Map.entry("sklearn.ensemble._forest.RandomForestRegressor", List.of(new Param("min_samples_leaf"), new Param("max_features"))),
+    Map.entry("sklearn.linear_model._coordinate_descent.ElasticNet", List.of(new Param("alpha", 0), new Param("l1_ratio"))),
+    Map.entry("sklearn.neighbors._unsupervised.NearestNeighbors", List.of(new Param(N_NEIGHBORS, 0))),
+    Map.entry("sklearn.neighbors._classification.KNeighborsClassifier", List.of(new Param(N_NEIGHBORS, 0))),
+    Map.entry("sklearn.neighbors._regression.KNeighborsRegressor", List.of(new Param(N_NEIGHBORS, 0))),
+    Map.entry("sklearn.svm._classes.NuSVC", List.of(new Param("nu"), new Param(KERNEL), new Param(GAMMA))),
+    Map.entry("sklearn.svm._classes.NuSVR", List.of(new Param(C), new Param(KERNEL), new Param(GAMMA))),
+    Map.entry("sklearn.svm._classes.SVC", List.of(new Param(C), new Param(KERNEL), new Param(GAMMA))),
+    Map.entry("sklearn.svm._classes.SVR", List.of(new Param(C), new Param(KERNEL), new Param(GAMMA))),
+    Map.entry("sklearn.tree._classes.DecisionTreeClassifier", List.of(new Param("ccp_alpha"))),
+    Map.entry("sklearn.tree._classes.DecisionTreeRegressor", List.of(new Param("ccp_alpha"))),
+    Map.entry("sklearn.neural_network._multilayer_perceptron.MLPClassifier", List.of(new Param("hidden_layer_sizes", 0))),
+    Map.entry("sklearn.neural_network._multilayer_perceptron.MLPRegressor", List.of(new Param("hidden_layer_sizes", 0))),
+    Map.entry("sklearn.preprocessing._polynomial.PolynomialFeatures", List.of(new Param("degree", 0), new Param("interaction_only"))));
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(CALL_EXPR, SklearnEstimatorHyperparametersCheck::checkEstimator);
+  }
+
+  private static void checkEstimator(SubscriptionContext ctx) {
+    CallExpression callExpression = (CallExpression) ctx.syntaxNode();
+
+    Symbol calleeSymbol = callExpression.calleeSymbol();
+
+    Optional.ofNullable(calleeSymbol)
+      .filter(callee -> callee.is(Kind.CLASS))
+      .map(ClassSymbol.class::cast)
+      .map(ClassSymbol::fullyQualifiedName)
+      .map(ESTIMATORS_AND_PARAMETERS_TO_CHECK::get)
+      .filter(parameters -> !isSetParamsCalled(callExpression))
+      .filter(parameters -> !isPartOfPipelineAndGridSearch(callExpression))
+      .filter(parameters -> isMissingAHyperparameter(callExpression, parameters))
+      .ifPresent(parameters -> ctx.addIssue(callExpression, MESSAGE));
+  }
+
+  private static boolean isMissingAHyperparameter(CallExpression callExpression, List<Param> parametersToCheck) {
+    return parametersToCheck.stream()
+      .map(param -> {
+        if (param.position.isPresent()) {
+          return TreeUtils.nthArgumentOrKeyword(param.position.get(), param.name, callExpression.arguments());
+        }
+        return TreeUtils.argumentByKeyword(param.name, callExpression.arguments());
```

Comment:
If you want to refactor, there should be a way with Optional to do that ( `orElseGet` or something like that). But it is not important at all